cargo-process.el

```diff
@@ -307,18 +307,29 @@ for error reporting."
       (json-read-from-string text)
     (error (user-error text))))
 
+(defun cargo-process--tramp-file-name-prefix (file-name)
+  "Return the TRAMP prefix for FILE-NAME.
+If FILE-NAME is not a TRAMP file, return an empty string."
+  (if (tramp-tramp-file-p file-name)
+      (let ((tramp-file-name (tramp-dissect-file-name file-name)))
+        (setf (nth 6 tramp-file-name) nil)
+        (tramp-make-tramp-file-name tramp-file-name))
+    ""))
+
 (defun cargo-process--workspace-root ()
   "Find the workspace root using `cargo metadata`."
   (when (cargo-process--project-root)
     (let* ((metadata-text (shell-command-to-string
                            (concat (shell-quote-argument cargo-process--custom-path-to-bin)
                                    " metadata --format-version 1 --no-deps")))
            (metadata-json (cargo-json-read-from-string metadata-text))
-           (workspace-root (cdr (assoc 'workspace_root metadata-json))))
+           (tramp-prefix (cargo-process--tramp-file-name-prefix (cargo-process--project-root)))
+           (workspace-root (concat tramp-prefix
+                                   (cdr (assoc 'workspace_root metadata-json)))))
       workspace-root)))
 
 (defun manifest-path-argument (name)
-  (let ((manifest-filename (cargo-process--project-root "Cargo.toml")))
+  (let ((manifest-filename (tramp-file-local-name (cargo-process--project-root "Cargo.toml"))))
```

Comment:
Where does this function come from? I can't run it locally.
```Emacs Lisp
(require 'tramp)
(tramp-file-local-name "~/somefile")
```

Emacs version: 26.3

Reply:
It seems this was added early 2020. Meaning you'd need a very recent version of Emacs for this to work https://github.com/emacs-mirror/emacs/commit/06364316e0998d6906b8a42d54102c5de4a54990

We might need to copy this function into cargo to prevent this package from breaking. (changing the name though)

Reply:
Ok, good to know. I can use a similar technique that I use to extract the TRAMP prefix.